dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/TrackedEntityInstanceController.java

```diff
@@ -287,6 +271,66 @@ public void getAttributeImage(
         }
     }
 
+    @GetMapping( value = "{teiId}/{attributeId}/files" )
+    public void getTrackedEntityAttributeFile(
+        @PathVariable( "teiId" ) String teiId,
+        @PathVariable( "attributeId" ) String attributeId, HttpServletResponse response )
+        throws WebMessageException
+    {
+        User currentUser = currentUserService.getCurrentUser();
+
+        org.hisp.dhis.trackedentity.TrackedEntityInstance tei = Optional
+            .ofNullable( instanceService.getTrackedEntityInstance( teiId ) )
+            .orElseThrow(
+                () -> new WebMessageException( notFound( "TrackedEntityInstance not found for ID " + teiId ) ) );
+
+        List<String> trackerAccessErrors = trackerAccessManager.canRead( currentUser, tei );
+
+        if ( !trackerAccessErrors.isEmpty() )
+        {
+            throw new WebMessageException(
+                unauthorized( "You're not authorized to access the TrackedEntityInstance with id: " + teiId ) );
+        }
+
+        Set<TrackedEntityAttributeValue> attributeValues = tei.getTrackedEntityAttributeValues();
+
+        Optional<TrackedEntityAttributeValue> optionalAttributeValue = attributeValues.stream()
+            .filter( att -> att.getAttribute().getUid().equals( attributeId ) )
+            .findFirst();
```

Comment:
Here you can use the same syntax as the tei before and use `orElseThrow` and you will not need to use `get()` later in the code

Reply:
resolved